File: backend/src/main/java/com/example/service/MemberService.java
package com.example.service;

import java.util.List;

import com.example.model.Member;

public interface MemberService {
	
    Member addMember(Member member);
    List<Member> getAllMembers();
    Member getMemberById(Integer memberId);
    Member getmemberByuserId(Integer memberId);
    Member updateMember(Integer memberId, Member updatedMember);
	String deleteMemberyId(Integer memberId);

}

File: Frontend/src/app/services/Member-service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from './baseUrl';

@Injectable({
  providedIn: 'root'
})
export class MemberService {
  private apiUrl = `${environment.apiBaseUrl}/api/members`
  constructor(private http : HttpClient) { }

  addMember(member: any) {
    return this.http.post(`${this.apiUrl}`, member);
  }
  getAllMembers() {
    return this.http.get(`${this.apiUrl}/getall`);
  }
  getMemberById(memberId: number) {
    return this.http.get(`${this.apiUrl}/${memberId}`);
  }
  getmemberByuserId(userId: number) {
    return this.http.get(`${this.apiUrl}/getByUserId/${userId}`);
  }
  updateMemberById(memberId: number, member: any) {
    return this.http.put(`${this.apiUrl}/${memberId}`, member);
  }
  deleteMemberById(memberId: number) {
    return this.http.delete(`${this.apiUrl}/${memberId}`);
  } 
}
